modules/protocol/unreal4.c

```diff
@@ -617,12 +617,41 @@ static void
 unreal_jupe(const char *server, const char *reason)
 {
 	struct service *svs;
+	static char sid[3+1];
+	int i;
 
 	server_delete(server);
 
 	svs = service_find("operserv");
 	sts(":%s SQUIT %s :%s", svs != NULL ? svs->nick : ME, server, reason);
-	sts(":%s SERVER %s 2 :%s", me.name, server, reason);
+
+	// dirty dirty make up some sid
+	if (sid[0] == '\0')
+		mowgli_strlcpy(sid, me.numeric, sizeof sid);
+	do
+	{
+		i = 2;
+		for (;;)
+		{
+			if (sid[i] == 'Z')
+			{
+				sid[i] = '0';
+				i--;
+
+				// eek, no more sids
+				if (i < 0)
+					return;
+
+				continue;
+			}
+			else if (sid[i] == '9')
+				sid[i] = 'A';
+			else sid[i]++;
+			break;
+		}
+	} while (server_find(sid));
```

Comment:
generally we like to be explicit about the NULL check

Reply:
I do happen to see a few potential problems here:

1. This algorithm will try generating `sid`s like `A00` after `9ZZ`.
2. `sid` is static, meaning it will eventually exceed `9ZZ` (even though nobody actually will run services _that_ long and jupe that many servers.)

I also wonder why don't reuse the original server `sid` (_if found_) and only generate a new `sid` when one is proactively jupe'ing servers before they were even introduced?

Reply:
this SID generation was directly stolen from the other protocol modules